framework/src/meshgenerators/GeneratedMeshGenerator.C

```diff
@@ -102,7 +102,8 @@ GeneratedMeshGenerator::GeneratedMeshGenerator(const InputParameters & parameter
 std::unique_ptr<MeshBase>
 GeneratedMeshGenerator::generate()
 {
-  std::unique_ptr<ReplicatedMesh> mesh = libmesh_make_unique<ReplicatedMesh>(comm(), 2);
+  // Have MOOSE construct the correct libMesh::Mesh object using Mesh block and CLI parameters.
+  auto mesh = _mesh->constructMeshBase();
```

Comment:
Ok - maybe... but I think `constructMesh()` really needs to take parameters about what's acceptable to be constructed.